nntrainer/tensor/manager.cpp

```diff
@@ -491,18 +498,21 @@ void Manager::initializeTensors(bool trainable) {
   if (!weights_initialized)
     initializeWeights();
 
-  // Allocate gradients
+  if (tensors_initialized)
+    return;
+
+  // Initialize gradients
   if (trainable)
     initializeGradients();
 
-  // Allocate shared derivative memory
+  // Initialize shared derivative memory
```

Comment:
isn't this allocate?

Reply:
no, this is `initialize`. Line 510 only sets `tensorDim` but does not allocate.

Reply:
Ah it's false. okay